### PR TITLE
support inheritance (without templates)

### PR DESCRIPTION
**Pull request by bbrehm**
Adds relevant field for inheritance. 

Is hopefully finished, not WIP, but I'll need to write end-to-end tests as well (and these should at least pass locally before we merge). Would profit from resolution of https://github.com/ShiftLeftSecurity/codepropertygraph/issues/429

Still missing feature: Base class is template. That would need grammar update.

### CODE REVIEW

File: src/test/scala/io/shiftleft/fuzzyc2cpg/astnew/AstToCpgTests.scala [hunk 1]
```diff
@@ -713,6 +717,46 @@ class AstToCpgTests extends WordSpec with Matchers {
...
+        |};
+      """.stripMargin
+    ) {
+
+      val derivedL = getTypeDecl("Derived")
+      derivedL.checkForSingle()
+
+      val derived = derivedL.head
+      // todo: clean up once /codepropertygraph/issues/429 is fixed
+      derived.valueMap().get("INHERITS_FROM_TYPE_FULL_NAME").asInstanceOf[Some[List[String]]].value shouldBe List("Base")
```
Comment (mpollmeier) [inline src/test/scala/io/shiftleft/fuzzyc2cpg/astnew/AstToCpgTests.scala]:
```suggestion
      derived.value[List[String]](NodeKeys.INHERITS_FROM_TYPE_FULL_NAME.name) shouldBe List("Base")
```

File: src/test/scala/io/shiftleft/fuzzyc2cpg/astnew/AstToCpgTests.scala [hunk 2]
```diff
@@ -713,6 +717,46 @@ class AstToCpgTests extends WordSpec with Matchers {
...
+        | int method(){return i;};
+        |};
+      """.stripMargin
+    ) {
+
+      val derivedL = getTypeDecl("Derived")
+      derivedL.checkForSingle()
+
+      val derived = derivedL.head
+      // todo: clean up once /codepropertygraph/issues/429 is fixed
```
Comment (mpollmeier) [inline src/test/scala/io/shiftleft/fuzzyc2cpg/astnew/AstToCpgTests.scala]:
```suggestion
```

File: src/test/scala/io/shiftleft/fuzzyc2cpg/astnew/AstToCpgTests.scala [hunk 3]
```diff
@@ -713,6 +717,46 @@ class AstToCpgTests extends WordSpec with Matchers {
...
```
Comment (mpollmeier) [inline src/test/scala/io/shiftleft/fuzzyc2cpg/astnew/AstToCpgTests.scala]:
```suggestion
```

File: src/test/scala/io/shiftleft/fuzzyc2cpg/astnew/AstToCpgTests.scala [hunk 4]
```diff
@@ -713,6 +717,46 @@ class AstToCpgTests extends WordSpec with Matchers {
...
+      derived.valueMap().get("INHERITS_FROM_TYPE_FULL_NAME").asInstanceOf[Some[List[String]]].value shouldBe List("OneBase", "TwoBase")
```
Comment (mpollmeier) [inline src/test/scala/io/shiftleft/fuzzyc2cpg/astnew/AstToCpgTests.scala]:
```suggestion
      derived.value[List[String]](NodeKeys.INHERITS_FROM_TYPE_FULL_NAME.name) shouldBe List("OneBase", "TwoBase")
```

File: src/main/java/io/shiftleft/fuzzyc2cpg/ast/declarations/ClassDefStatement.java [hunk 5]
```diff
@@ -1,18 +1,21 @@
...
 import io.shiftleft.fuzzyc2cpg.ast.logical.statements.Statement;
 import io.shiftleft.fuzzyc2cpg.ast.walking.ASTNodeVisitor;
+import java.util.LinkedList;
 
 public class ClassDefStatement extends Statement {
 
   public Identifier identifier = new DummyIdentifierNode();
   public CompoundStatement content = new CompoundStatement();
   protected TemplateParameterList templateParameterList;
+  public LinkedList<Identifier> baseClasses = new LinkedList<Identifier>();
```
Comment (GlassAndOneHalf) [inline src/main/java/io/shiftleft/fuzzyc2cpg/ast/declarations/ClassDefStatement.java]:
```suggestion
  public LinkedList<Identifier> baseClasses = new LinkedList<>();
```

File: src/main/antlr4/io/shiftleft/fuzzyc2cpg/Common.g4 [hunk 6]
```diff
@@ -146,6 +146,7 @@ class_key: 'struct' | 'class' | 'union' | 'enum';
 class_def: template_decl? class_key gcc_attribute? class_name? template_args? base_classes? OPENING_CURLY {skipToEndOfObject(); } ;
 class_name: identifier;
 base_classes: ':' base_class (',' base_class)*;
+// todo: base_class: VIRTUAL? access_specifier? identifier ('<' template_args '>')?;
```
Comment (fabsx00) [inline src/main/antlr4/io/shiftleft/fuzzyc2cpg/Common.g4]: Let's remove this and rather create a ticket.

File: src/main/java/io/shiftleft/fuzzyc2cpg/parser/modules/CModuleParserTreeListener.java [hunk 7]
```diff
@@ -177,6 +185,7 @@ public void enterClass_name(ModuleParser.Class_nameContext ctx) {
     builder.setName(ctx);
   }
 
+
```
Comment (fabsx00) [inline src/main/java/io/shiftleft/fuzzyc2cpg/parser/modules/CModuleParserTreeListener.java]: Remove

**Pull request merged**